src/app/core/app-config/app-config.ts

```diff
@@ -34,46 +35,55 @@ import { IAppConfig } from "./app-config.model";
-
-  /** fallback file location of the config that is part of the project already if the "real" config file isn't found */
-  private static readonly DEFAULT_CONFIG_FILE = "assets/config.default.json";
+  /** Path for the reverse proxy that forwards to the database - configured in `proxy.conf.json` and `default.conf` */
+  static readonly DB_PROXY_PREFIX = "/db";
+  /** Name of the database that is used */
+  static readonly DB_NAME = "app";
+  /** Demo and session can be persisted in local storage */
+  static readonly DEMO_LOCAL_STORAGE_KEY = "demo_mode";
+  static readonly SESSION_LOCAL_STORAGE_KEY = "session_type";
+  /** Whether the app is running in demo mode */
+  static DEMO_MODE = false;
+  /** The session type that is used */
+  static SESSION_TYPE = SessionType.synced;
 
   /**
-   * Load the config file into the `AppConfig.settings` so they can be used synchronously anywhere in the code after that.
+   * Initializes static settings through the URL params, the local storage or the environment.
+   *
+   * The settings are checked and used in the following order:
+   * 1. URL params
+   * 2. Local storage
+   * 3. Environment
    *
-   * If the config file does not exist, uses the default config as a fallback.
+   * If params are found in the URL query params, then they are persisted in the local storage.
    */
-  static load(): Promise<IAppConfig> {
-    return this.loadAppConfigJson(this.CONFIG_FILE).catch(() =>
-      this.loadAppConfigJson(this.DEFAULT_CONFIG_FILE)
-    );
+  static initSettings() {
+    const params = new URLSearchParams(location.search);
+    this.DEMO_MODE = this.isDemoMode(params);
+    this.SESSION_TYPE = this.getSession(params);
   }
 
-  /**
-   * Load the given file and set it as AppConfig.settings.
-   *
-   * This requires a HTTP request because the "assets" folder including the config.json is on our server
-   * while this javascript code is executed in the users browser.
-   *
-   * @param jsonFileLocation The file path of the json file to be loaded as config
-   */
-  private static loadAppConfigJson(
-    jsonFileLocation: string
-  ): Promise<IAppConfig> {
-    return fetch(jsonFileLocation)
-      .then((result) => result.json())
-      .then((result: IAppConfig) => (AppConfig.settings = result))
-      .catch((response: any) => {
-        throw new Error(
-          `Could not load file '${jsonFileLocation}': ${JSON.stringify(
-            response
-          )}`
-        );
-      });
+  private static isDemoMode(params: URLSearchParams): boolean {
+    const demoParam = params.get("demo");
+    const demoStorage = localStorage.getItem(this.DEMO_LOCAL_STORAGE_KEY);
+    if (demoParam) {
+      localStorage.setItem(this.DEMO_LOCAL_STORAGE_KEY, demoParam);
+      return demoParam === "true";
+    } else if (demoStorage) {
+      return demoStorage === "true";
+    }
+    return environment.demo_mode;
+  }
+
+  private static getSession(params: URLSearchParams): SessionType {
```

Comment:
we could generalize this and have only one method "loadSettingFromUrlLocalOrEnvironment(params, key)". If we use the same key for the query param, local storage and environment.ts variable that might make it easier to follow anyway.